GeoLib/Grid.h

```diff
@@ -384,29 +376,42 @@ void Grid<POINT>::getGridCoords(T const& pnt, std::size_t* coords) const
 			}
 		}
 	}
+	return coords;
+}
+
+template <typename POINT>
+template <typename P>
+std::array<double,6> Grid<POINT>::getPointCellBorderDistances(P const& p,
+	std::array<std::size_t,3> const& coords) const
+{
+	std::array<double,6> dists;
+	dists[0] = std::abs(p[2]-this->_min_pnt[2] + coords[2]*_step_sizes[2]); // bottom
+	dists[5] = std::abs(p[2]-this->_min_pnt[2] + (coords[2]+1)*_step_sizes[2]); // top
+
+	dists[1] = std::abs(p[1]-this->_min_pnt[1] + coords[1]*_step_sizes[1]); // front
+	dists[3] = std::abs(p[1]-this->_min_pnt[1] + (coords[1]+1)*_step_sizes[1]); // back
+
+	dists[4] = std::abs(p[0]-this->_min_pnt[0] + coords[0]*_step_sizes[0]); // left
+	dists[2] = std::abs(p[0]-this->_min_pnt[0] + (coords[0]+1)*_step_sizes[0]); // right
+	return dists;
 }
 
 template <typename POINT>
 template <typename P>
 POINT* Grid<POINT>::getNearestPoint(P const& pnt) const
 {
-	std::size_t coords[3];
-	getGridCoords(pnt, coords);
+	std::array<std::size_t,3> coords(getGridCoords(pnt));
 
-	double sqr_min_dist (MathLib::sqrDist(this->_min_pnt, this->_max_pnt));
+	double sqr_min_dist(MathLib::sqrDist(this->_min_pnt, this->_max_pnt));
 	POINT* nearest_pnt(nullptr);
 
-	double dists[6];
-	getPointCellBorderDistances(pnt, dists, coords);
+	std::array<double,6> dists(getPointCellBorderDistances(pnt, coords));
 
 	if (calcNearestPointInGridCell(pnt, coords, sqr_min_dist, nearest_pnt)) {
 		double min_dist(sqrt(sqr_min_dist));
-		if (dists[0] >= min_dist
-						&& dists[1] >= min_dist
-						&& dists[2] >= min_dist
-						&& dists[3] >= min_dist
-						&& dists[4] >= min_dist
-						&& dists[5] >= min_dist) {
+		if (dists[0] >= min_dist && dists[1] >= min_dist
+			&& dists[2] >= min_dist && dists[3] >= min_dist
+			&& dists[4] >= min_dist && dists[5] >= min_dist) {
 			return nearest_pnt;
 		}
 	} else {
```

Comment:
The else branch seems not to be covered by the tests (and was not before). So this is just a remark.
@TomFischer can you extend the tests, such the else branch is taken in the feature?